src/Graphing.hs

```diff
@@ -61,6 +62,14 @@ filter f gr = gr { graphingDirect = direct', graphingAdjacent = adjacent' }
 empty :: Graphing ty
 empty = Graphing S.empty AM.empty
 
+-- | Strip a given item from the direct set, promote its immediate children to direct items
```

Comment:
doc here is silghtly confusing -- there isn't really a 'given item'  here, and it'd apply to all direct dependencies

Reply:
This came from my first attempt, I'll rewrite it.